clients/lsp-javascript.el

```diff
@@ -841,24 +841,18 @@ name (e.g. `data' variable passed as `data' parameter)."
                                (lsp-point-to-position end)))
        (lambda (res)
          (lsp--remove-overlays 'lsp-javascript-inlay-hint)
-         (let ((hints (lsp-get res :inlayHints)))
-           (unless (seq-empty-p hints)
-             (overlay-recenter
-              (-let* (([hint] hints)
-                      ((&javascript:InlayHint :position) hint))
-                (lsp--position-to-point position))))
-           (-each hints
-             (lambda (hint)
-                 (-let* (((&javascript:InlayHint :text :position :kind :whitespace-before? :whitespace-after?) hint)
-                         (pos (lsp--position-to-point position))
-                         (overlay (make-overlay pos pos nil 'front-advance 'end-advance)))
-                   (overlay-put overlay 'lsp-javascript-inlay-hint t)
-                   (overlay-put overlay 'before-string
-                                (format "%s%s%s"
-                                        (if (and whitespace-before? (not (string= kind lsp/javascript-inlay-hint-kind-type-hint))) " " "")
-                                        (propertize (lsp-javascript-format-inlay text kind)
-                                                    'font-lock-face (lsp-javascript-face-for-inlay kind))
-                                        (if whitespace-after? " " ""))))))))
+         (dolist (hint res)
+           (-let* (((&javascript:InlayHint :label :position :kind :padding-left? :padding-right?) hint)
+                   (pos (lsp--position-to-point position))
+                   (overlay (make-overlay pos pos nil 'front-advance 'end-advance)))
+              (when (stringp label)
+                (overlay-put overlay 'lsp-javascript-inlay-hint t)
+                (overlay-put overlay 'before-string
+                             (format "%s%s%s"
+                                     (if padding-left? "" "")
```

Comment:
both legs "" ?

Reply:
It looks ugly with `" " ""`
```js
  const bar : string ...
```
vs 
```js
 const bar: string ...
```
I'm not a typescript/javascript dev, so I don't know what is preferred.